### DISCUSSION TITLE
How get the RAW value of attributes ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2313] [Q&A] [answered]
Hi, 

I try to read the `dataset` of an element, but this is being always parsed ! and it should not.. any ideas ?

HTML code
```html
<div id="test" data-something="[ 'test', 'mc&#39;Donnel' ]"
```
Brython
```python
print( document['test'].dataset.something)
```
output
> [ 'test', 'mc'Donnel' ]

and so I can't manipulate this wrong formatted list :'( any ideas ?

WebBrowser console:
> invalid syntax. Perhaps you forgot a comma?

BTW" I'm running Brython 3.11.2

**Comment by denis-migdal** [comment]
Did you try with `attrs` ?

```python
print( document['test'].attrs["data-something"])
```

**Reply by BobSquarePants** [reply]
Hi @denis-migdal ,

Yes, i've tried few things

```python
print( document['test'].attrs["data-something"])
print( document['test'].getAttribute('data-something'))
```

but every time it get parsed :'(

**Reply by denis-migdal** [reply]
I tested on Brython 3.12, we do get a string. It doesn't seem to be parsed :
```python
from browser import document
from browser import console

print( "3", document['test'].dataset.something, type( document['test'].dataset.something ) )
```
```
3 [ 'test', 'mc'Donnel' ] <class 'str'>
```
https://brython.info/tests/editor.html?lang=fr&code=from%20browser%20import%20document%0Afrom%20browser%20import%20console%0A%0Aprint%28%20%223%22%2C%20document%5B'test'%5D.dataset.something%2C%20type%28%20document%5B'test'%5D.dataset.something%20%29%20%29

In a string, `'` and `&#39;` are the same thing. If you want to escape it : `\'`.
Or if you build it from strings, the easier way is to use `"` instead of `'` : `f"[\"{val1}\", \"{val2}\"]"`.

**Reply by BobSquarePants** [reply]
Thanks to have tried !

actually the data come from user, and therefor fully escaped with --> https://palletsprojects.com/p/markupsafe/

in a string  `'` and `&#39;` are not the same ! but in html ( if parsed yes )

and I can't escape `'` with like `\'` as those simple quote are encoded as `&#39;`

**Comment by denis-migdal** [comment]
> in a string `'` and `&#39;` are not the same !

Indeed, my bad.

> but in html ( if parsed yes )

As you do `data-something="[ 'test', 'mc&#39;Donnel' ]"`, it'll be "parsed" by HTML, and `&#39;` will be replaced by `'` when you'll try to access it in Brython or in JS.

~~Why not using `JSON.stringify([val1, val2])` ?~~

**Reply by denis-migdal** [reply]
Note that `<div data-something="[ 'test', 'mc&#39;Donnel' ]">` ("parsed"), isn't the same thing as doing `mydiv.dataset.something="[ 'test', 'mc&#39;Donnel' ]"` (not "parsed").

**Comment by BobSquarePants** [comment]
indeed,

I've just discover than even with a value of `[" it's me ","something else"]` it crash too (can't eval the string) because of the `'` so weird !! as is surrounded by `"`

**Reply by denis-migdal** [reply]
Use `JSON.stringify()` and `JSON.parse()` instead of using `eval()`, it'll take care of all quote escaping for you.

**Reply by denis-migdal** [reply]
Okay, got it.

If you write in HTML : `f"<div data-something="{str.replace("&", "&amp;")}">"`, it'll make it so you get the `&#39;` in JS.
But I think the best way is to use `JSON.stringify()` and `JSON.parse()` instead of `eval()` (which seems a kind of strange and dangerous way to proceed).

**Reply by BobSquarePants** [reply, marked as answer]
We were all ~wrong,

I finally found were was the bug. To generate my html I'm using Flask/Jinja.

when my template had an extension `.htm` no problems at all but when I changed to `.jinja` then that trigger this odd behavior !!! I'm glad I've found, but OMG I could still dig for a long time...

Cheers.
